### PR TITLE
Adapt storage monitoring to new storage structure

### PR DESCRIPTION
**Pull request by ahmad-elbatanouni**

### CODE REVIEW

**Comment by ahmad-elbatanouni** [comment]
[Adapt storage monitoring to new storage structure](https://trello.com/c/xB6FwP9P/122-adapt-storage-monitoring-to-new-storage-structure)

File: functions/src/modules/Files.functions.ts [hunk 1]
```diff
@@ -31,36 +33,34 @@ export const importFilesFromStorage = functions.storage
...
-  .bucket(`original${functions.config().storage['root-domain']}`)
+export const handleOriginalFileDeletion = functions.storage
+  .bucket(originalBucket)
   .object()
   .onDelete(async object => {
     const filePath = object.name;
     const nameParts = checkValidFile(filePath);
-    if (!nameParts) return -1;
+    if (!nameParts) {
+      console.warn(`File: "${filePath}" was deleted.`);
```
Comment (anantakrishna) [inline functions/src/modules/Files.functions.ts]: Warning is not about the fact that the file was deleted, but about the fact that we ignore it for some reason.

File: functions/src/modules/Files.functions.ts [hunk 2]
```diff
@@ -31,36 +33,34 @@ export const importFilesFromStorage = functions.storage
     const list = helpers.extractListFromFilename(fileName);
 
     const fileRef = db.ref(`/files/${list}/${fileName}`);
+    const file = (await fileRef.once('value')).val();
 
     fileRef.set({
-      mp3Uploaded: isValidFile[3] ? true : false,
-      flacUploaded: isValidFile[5] ? true : false,
+      mp3Uploaded: isValidFile[3] ? true : file.mp3Uploaded,
+      flacUploaded: isValidFile[5] ? true : file.flacUploaded
```
Comment (anantakrishna) [inline functions/src/modules/Files.functions.ts]: Use the same approach that you use below on line 59.

File: functions/src/modules/Files.functions.ts [hunk 3]
```diff
@@ -89,10 +89,11 @@ export const syncStorageToDB = functions.https
       const list = helpers.extractListFromFilename(fileName);
 
       const fileRef = db.ref(`/files/${list}/${fileName}`);
+      const file = (await fileRef.once('value')).val();
 
       fileRef.update({
-        mp3Uploaded: parts[3] ? true : false,
-        flacUploaded: parts[5] ? true : false,
+        mp3Uploaded: parts[3] ? true : file.mp3Uploaded,
+        flacUploaded: parts[5] ? true : file.flacUploaded,
```
Comment (anantakrishna) [inline functions/src/modules/Files.functions.ts]: Use approach from line 59.

File: functions/src/modules/Files.functions.ts (hunk 1 above)
Comment (ahmad-elbatanouni) [reply functions/src/modules/Files.functions.ts]: This function is triggered when files are deleted, there's no way to find out the reason that caused the file to be deleted.

**Comment by ahmad-elbatanouni** [comment]
There's a problem with the `master` branch ... the `index.ts` file uses a `ReportingSync` module that neither has a corresponding file nor an import.

**Comment by anantakrishna** [comment]
> There's a problem with the `master` branch ... the `index.ts` file uses a `ReportingSync` module that neither has a corresponding file nor an import.

You should merge master into your branch and resolve the conflicts. You removed `ReportingSync` in this branch.

**Comment by anantakrishna** [comment]
Never mind, I have merged myself.

**Pull request merged**